### ISSUE TITLE
Issue running wasm compilation out-of-the-box

### ISSUE THREAD

**Issue by jonbae** [CosmWasm/cw-plus#545]
when running the docker command to compile
```
docker run --rm -v "$(pwd)":/code \
  --mount type=volume,source="$(basename "$(pwd)")_cache",target=/code/target \
  --mount type=volume,source=registry_cache,target=/usr/local/cargo/registry \
  cosmwasm/workspace-optimizer:0.12.3
```
I get a error of 
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 20, kind: NotADirectory, message: "Not a directory" }', src/main.rs:58:14
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

I have done `npm install ` in the root directory on all the necessary packages for base-helpers.ts to run. 

I have also run the command `RUSTFLAGS="-C link-arg=-s" cargo build --release --target=wasm32-unknown-unknown --locked` as suggested to debug but all of the contracts compile fine. 

currently stuck and would appreciate help. 

Rust: rustc 1.56.1
OS: windows subsystem for linux

**Comment by wojtczal** [comment]
I assume you're seeing a log such as `Building "contracts/base-helpers.ts" ...` right before what you pasted as an error. This happens because in the `contracts` directory which rust is trying to compile there is a typescript file. This is a recent change made by @findolor - maybe he can shed more light on it ;)

**Comment by findolor** [comment]
Thanks for bringing this up! You are right this is expecting every item to be a directory in ```contracts```.
Opened a PR to fix this one.

**Comment by webmaster128** [comment]
Thank you for the report. This is a missing _is directory_ check in https://github.com/CosmWasm/rust-optimizer/blob/v0.12.3/build_workspace/src/main.rs#L28-L32

@maurolacy could you look into this?

**Comment by ethanfrey** [comment]
I believe this has since been fixed.
Both in a more robust rust-optimiser as well as in removing the base_helpers.ts file.

Can this be closed?

**Comment by webmaster128** [comment]
True, fixed in 0.12.4 with this function: https://github.com/CosmWasm/rust-optimizer/blob/v0.12.4/build_workspace/src/main.rs#L22-L31.

I think it can be closed once all references to rust-optimizer in this project are updated.
